src/smolagents/logger.py

```diff
@@ -0,0 +1,284 @@
+import json
+from dataclasses import asdict, dataclass
+from enum import IntEnum
+from typing import TYPE_CHECKING, Any, Dict, List
+
+from rich.console import Console
+from rich.rule import Rule
+from rich.syntax import Syntax
+
+from smolagents.models import MessageRole
+from smolagents.utils import AgentError, make_json_serializable
+
+
+if TYPE_CHECKING:
+    from smolagents.models import ChatMessage
+
+
+console = Console()
+
+
+@dataclass
+class Message:
+    role: MessageRole
+    content: str | list[dict]
+
+    def dict(self):
+        return asdict(self)
```

Comment:
If you derive from `TypedDict` (instead of making it a dataclass), there is no need of the `.dict()` method.

```python
from typing import TypedDict
```
```suggestion
class Message(TypedDict):
    role: MessageRole
    content: str | list[dict]
```
Afterwards, you can use, e.g:
```python
memory.append(Message(role=MessageRole.SYSTEM, content=self.agent_memory"))

Reply:
Oh nice!